File: frontend/src/service/EmployeeService.js
import axios from "axios";

const BASE_URL= "http://localhost:8080/employee";

class EmployeeService
{
    getAllEmployee()
    {
        return axios.get(BASE_URL);
    }

    saveEmployee(employeeData)
    {
        return axios.post(BASE_URL, employeeData);
    }

    updateEmployee (id, employeeData)
    {
        return axios.put(`${BASE_URL}/${id}`, employeeData);
    }

    getEmployeeById (id)
    {
        return axios.get(`${BASE_URL}/${id}`);
    }

    deleteEmployee (id)
    {
        return axios.delete(BASE_URL + "/" + id);
    }
}

export default new EmployeeService();

File: backend/src/main/java/com/employeeAPI/management/service/EmployeeService.java
package com.employeeAPI.management.service;

import com.employeeAPI.management.model.Employee;
import com.employeeAPI.management.repository.EmployeeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class EmployeeService implements EmployeeServiceInterface
{
    @Autowired
    private EmployeeRepository employeeRepository;

    @Override
    public Employee saveEmployee(Employee employee)
    {
        return employeeRepository.save(employee);
    }

    @Override
    public Optional<Employee> getEmployeeById(int id)
    {
        return employeeRepository.findById(id);
    }

    @Override
    public List<Employee> getAllEmployee()
    {
        return employeeRepository.findAll();
    }

    @Override
    public Employee updateEmployee(int id, Employee employee)
    {
        Employee empToUpdate = employeeRepository.findById(id).orElseThrow();

        empToUpdate.setFirstName(employee.getFirstName());
        empToUpdate.setLastName(employee.getLastName());
        empToUpdate.setEmail(employee.getEmail());

        return employeeRepository.save(empToUpdate);
    }

    @Override
    public void deleteEmployee(int id)
    {
        employeeRepository.deleteById(id);
    }
}
